viewer/cache.hpp

```diff
@@ -4,6 +4,8 @@
 #include <hawktracer/base_types.h>
 #include <hawktracer/parser/event.hpp>
 
+#include <boost/optional.hpp>
```

Comment:
Hmm I forgot std::optional was only introduced in c++17, I'm not sure @loganek would want to introduce a boost dependency to the project.  You could perhaps return a pointer instead, or change the signature of the function to 

    bool get_data(HT_EventKlassId klass_id, std::vector<EventRef>& event);

and use event as an output parameter and return true if the id exists and false otherwise...  I didn't mean to overcomplicate your pull request though so if you think it was better how it was just revert those changes.